integrationtests/src/test/java/io/kroxylicious/proxy/multitenant/MultiTenantIT.java

```diff
@@ -435,4 +444,33 @@ private static <K, V> Predicate<ConsumerRecord<K, V>> matchesRecord(final String
             return Objects.equals(rec.topic(), expectedTopic) && Objects.equals(rec.key(), expectedKey) && Objects.equals(rec.value(), expectedValue);
         };
     }
+
+    private static class PartitionAssignmentAwaitingRebalanceListener<K, V> implements ConsumerRebalanceListener {
+        private final CompletableFuture<Object> future = new CompletableFuture<>();
+        private final KafkaConsumer<K, V> consumer;
+
+        public PartitionAssignmentAwaitingRebalanceListener(KafkaConsumer<K, V> consumer) {
+            this.consumer = consumer;
+        }
+
+        @Override
+        public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+
+        }
+
+        @Override
+        public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+            future.complete(null);
+        }
+
+        public void awaitAssignment(Duration timeout) throws TimeoutException {
+            var deadline = Instant.now().plus(timeout);
+            do {
+                consumer.poll(Duration.ofMillis(50));
+            } while (!future.isDone() && Instant.now().isBefore(deadline));
+            if (!future.isDone()) {
+                throw new TimeoutException("timed out awaiting consumer assignment");
+            }
```

Comment:
```suggestion
            Awaitility.await().atMost(timeout).until(() -> { 
                consumer.poll(Duration.ofMillis(50));
                return assigned.get();
            });
```
or maybe 
```suggestion
            Awaitility.await().atMost(timeout).conditionEvaluationListener(new ConditionEvaluationListener() {
                @Override
                public void beforeEvaluation(StartEvaluationEvent startEvaluationEvent) {
                    consumer.poll(Duration.ofMillis(50));
                }

                @Override
                public void conditionEvaluated(EvaluatedCondition condition) {
                    //Shrug
                }
            }).untilTrue(reassigned);
```

Not sure if the second option helps readability directly here but I think it could be expanded to make things more generically useful later.